### ISSUE TITLE
Add checks for OpenJDK Style §3.5 - Modifiers

### ISSUE THREAD

**Issue by achyuth8055** [checkstyle/checkstyle#19629]
Parent issue: #19604

### Description

Add Checkstyle checks for all rules under §3.5 - Modifiers of the OpenJDK Java Style Guidelines v6.

### Rules covered

From [§3.5 - Modifiers](https://checkstyle.org/styleguides/openjdk-java-style-v6/openjdk-styleguide.html#s3.5-modifiers):

1. Modifiers should go in the order: access modifier, `abstract`, `static`, `final`, `transient`, `volatile`, `default`, `synchronized`, `native`, `strictfp` — `ModifierOrder` Check
2. Modifiers should not be written out when they are implicit (e.g., interface methods should not be declared `public` or `abstract`, nested enums/interfaces should not be declared `static`) — `RedundantModifier` Check
3. Method parameters and local variables should not be declared `final` unless it improves readability — cannot be enforced (subjective guideline)
4. Fields should be declared `final` unless there is a compelling reason to make them mutable — cannot be enforced (subjective guideline)

### Checks to add to `openjdk_checks.xml`

TreeWalker-level:

```xml
<module name="ModifierOrder"/>

<module name="RedundantModifier"/>

**Comment by romani** [comment]
@achyuth8055 , thanks a lot for interest to help on this issue,before you start doing it, please finish your hanging PRs on trim of Inputs

**Comment by vivek-0509** [comment]
Issue looks good, mentioned  check fully covers the enforceble rules

**Comment by romani** [comment]
fix merged.
